wiring/inc/spark_wiring_system.h

```diff
@@ -435,6 +437,7 @@ class SystemClass {
 
 private:
     SleepResult sleepResult_;
+    SystemSleepResult systemSleepResult_;
```

Comment:
Ideally we should also fill this structure when waking up from deep sleep (same as with `SleepResult`). Does it make sense to simply use `SystemSleepResult` everywhere instead and provide `operator SleepResult()` not to lose compatibility?

Reply:
I'd like to introduce `SystemSleepResult` as the private member of `SleepResult` and keep the methods of it acting as wrappers of the `SystemSleepResult` methods. Sounds good?

Reply:
It still needs to be castable to `SleepResult` if we are to change the return type of Sleep 1.0 functions, because the user code may have:
```c++
SleepResult result = System.sleep(XXX);
```

Reply:
Can't we just `return SleepResult(System.systemSleepResult_);`?

Reply:
Yes, as long as `System.systemSleepResult_` is convertible to `SleepResult`.